### PR TITLE
rgw: loadgen fix generate random object name rgw crash issue

### PR DESCRIPTION
**Pull request by ZVampirEM77**
Fixes: http://tracker.ceph.com/issues/22006

Signed-off-by: Enming Zhang <enming.zhang@umcloud.com>

### CODE REVIEW

**Comment by theanalyst** [comment]
can you explain the crash you're seeing?

**Comment by ZVampirEM77** [comment]
@theanalyst The definition of `gen_rand_alphanumeric ` is
`int gen_rand_alphanumeric(CephContext *cct, char *dest, int size);`,
`cct->random()->get_bytes(dest, size);` will be called in `gen_rand_alphanumeric` 
in lastest master branch. If calling `gen_rand_alphanumeric` with `cct = NULL` as `RGWLoadGenProcess::run()` doing before, the rgw will crash.

Actually, I do not verify whether rgw will crash or not during using loadgen with generating random object ID. But I have met the rgw crash issue during PutLC without ID tag and I have
fix the issue in my another PR https://github.com/ceph/ceph/pull/18534. I think it is a same issue.

**Comment by theanalyst** [comment]
Ah now I understand, there was a PR that went in recently that requires ctx to be a part of gen_random_bytes, this looks right, but I'm not familiar with the loadgen interface

File: src/rgw/rgw_loadgen_process.cc [hunk 1]
```diff
@@ -60,7 +60,7 @@ void RGWLoadGenProcess::run()
 
   for (i = 0; i < num_objs; i++) {
     char buf[16 + 1];
-    gen_rand_alphanumeric(NULL, buf, sizeof(buf));
+    gen_rand_alphanumeric(cct, buf, sizeof(buf));
```
Comment (cbodley) [inline src/rgw/rgw_loadgen_process.cc]: there's a call to `append_rand_alpha(NULL, bucket, bucket, 16);` above that needs a cct also

Comment (ZVampirEM77) [reply src/rgw/rgw_loadgen_process.cc]: @cbodley Done~

**Pull request merged**